src/utils.ts

```diff
@@ -132,3 +132,41 @@ export const isDefined = (value: any): boolean => value !== void 0;
 
 export const isMatchingPath = (path: string, criteria: string | RegExp) =>
     isRegex(criteria) ? !!path.match(criteria as RegExp) : checkGlob(path, criteria as string);
+
+export const isExpanded = (path: string, state: JsonViewerState): boolean => state.expanded[path];
+
+export const isCollapsed = (path: string, state: JsonViewerState): boolean => !state.expanded[path];
+
+export const isDeepestVisible = (node: HTMLElement): boolean =>
```

Comment:
i'm not sure i follow this function. is the idea that "there are no `part`s within this node other than itself?"

Reply:
Nope, every node contains 2 parts (node should always be an li here). This is just checking that none of the children of the node have `part=object` 

There are 3 "types" of elements in my head: 
1. Primitive 
2. Expanded Object (has a child with `[part=object]`)
3. Collapsed Object


There are definitely other ways to check this, I'm just using a querySelector since I was querySelector happy instead of checking `[1]` does something like `![...node.children].some(element => element.getAttribute('part') === 'object')` make that more clear?

Reply:
ok this was the piece i wasn't fully grokking: `part=object` only exists on expanded objects!